playwright/src/main/java/com/microsoft/playwright/impl/BrowserTypeImpl.java

```diff
@@ -73,6 +73,7 @@ private Browser connectImpl(String wsEndpoint, ConnectOptions options) {
           slowMo = Duration.ofMillis(options.slowMo.intValue());
         }
       }
+      headers.put("User-Agent", getUserAgent());
```

Comment:
User-agent passed in the headers parameter should override default value. So this statement should probably moved up.

Reply:
fixed

Reply:
Have added tests.